### PR TITLE
Support 'iris_grib'; deprecate 'iris.fileformats.grib'.

### LINKED ISSUE #2019: Deprecate iris.FUTURE.strict_grib_load
This is redundant thanks to iris-grib.

### PR DESCRIPTION
**Pull request by pp-mo**
Will not check out until we actually have a functional "iris_grib".
Even then the examples need a further tweak.

Closes #2019

### CODE REVIEW

File: docs/iris/example_code/General/polar_stereo.py [hunk 1]
```diff
@@ -15,9 +15,17 @@
...
+    # intended for future Iris versions.
+    iris.FUTURE.external_grib_support = True
+
+    # What if anything is needed here for the first version of iris_grib ?
+    # - we don't actually know yet ...
+    # What we might need in future...
+    #    import iris_grib
+    #    iris_grib.FUTURE.strict_grib_load = True
+    # FOR NOW: a nasty kluge, as setting this is now *itself* deprecated.
+    iris.FUTURE.__dict__['strict_grib_load'] = True
```
Comment (pp-mo) [inline docs/iris/example_code/General/polar_stereo.py]: Note: when iris_grib is available, this becomes "**iris_grib**.FUTURE.\_\_dict__\['strict_grib_load'] = True"

File: docs/iris/example_code/General/polar_stereo.py [hunk 2]
```diff
@@ -15,9 +15,17 @@
...
-    iris.FUTURE.strict_grib_load = True
+    # Enable a future option, to ensure that the grib load uses the mechanism
+    # intended for future Iris versions.
+    iris.FUTURE.external_grib_support = True
+
+    # What if anything is needed here for the first version of iris_grib ?
+    # - we don't actually know yet ...
+    # What we might need in future...
+    #    import iris_grib
+    #    iris_grib.FUTURE.strict_grib_load = True
```
Comment (pp-mo) [inline docs/iris/example_code/General/polar_stereo.py]: Confirmed : this *is* what we should have here
See [iris-grib#9](https://github.com/SciTools/iris-grib/pull/9)

**Comment by rhattersley** [comment]
Instead of the FUTURE switch we should just check for the presence of `iris_grib`.

File: docs/iris/example_code/General/polar_stereo.py (hunk 2 above)
Comment (pp-mo) [inline docs/iris/example_code/General/polar_stereo.py]: Oh no it isn't : [updated iris-grib#9, comment](https://github.com/SciTools/iris-grib/pull/9#issuecomment-219436588)

**Comment by pp-mo** [comment]
Replaced by #2084

**Pull request closed without merging**